### ISSUE TITLE
NerDLApproach TestDataset requires embeddings column, required to be in parquet

### ISSUE THREAD

**Issue by ethnhll** [JohnSnowLabs/spark-nlp#13070]
## Summary
For training an NER model, there is a setter for NerDLApproach that provides a test dataset for evaluation (`setTestDataset`). This setter requires the dataset to be in parquet format, and requires an embeddings column in the data to be usable by NerDLApproach.

## Description

This is a bit of a mixed bag of an issue, as it could be considered a documentation request, a fix (unintended behavior), or a feature request.

The requirement for the test dataset for NerDLApproach to be in parquet isn't documented. 
```
    testDataset
        Path to test dataset. If set used to calculate statistic on it during
        training.
```

Providing a path to a CoNLL plain text file for the test dataset causes the training to fail with this error.
```java
Could not read footer: java.io.IOException: Could not read footer for file ......... at parquet.hadoop.ParquetFileReader.readAllFootersInParallel(ParquetFileReader.java:248) at org.apache.spark.sql.execution.datasources.parquet.ParquetRelation$$anonfun$28.apply(ParquetRelation.scala:750)
```

If the CoNLL file is in parquet, providing the test dataset still causes training to fail, as there is no `embeddings` column in the CoNLL data. The test dataset requiring an embeddings column is undocumented.
```python
pyspark.sql.utils.AnalysisException: Column 'embeddings' does not exist. Did you mean one of the following? [label, sentence, token, document, pos, text];
'Project [label#154, sentence#151, token#152, 'embeddings]
+- Relation [text#149,document#150,sentence#151,token#152,pos#153,label#154] parquet
```

## Expected Behavior

I would expect to be able to provide a path to a plain-text CoNLL file for a test dataset without failure.
I also would expect to not need to provide embeddings for the test dataset, as the embeddings are not what is being evaluated when training an NER model.

Perhaps most importantly, these caveats on params should be documented in the API docs.

## Current Behavior
See the description above.

## Steps to Reproduce
```python
.....
    ner_approach = NerDLApproach() \
        .setInputCols([ColumnLabels.SENTENCE, ColumnLabels.TOKEN, ColumnLabels.EMBEDDINGS]) \
        .setLabelColumn(ColumnLabels.LABEL) \
        .setOutputCol(ColumnLabels.NER) \
        .setOutputLogsPath('./ner_logs') \
        .setGraphFolder("./ner_graphs") \
        .setEnableMemoryOptimizer(True) \
        .setUseBestModel(True) \
        .setTestDataset("PATH/TO/PLAINTEXT/CONLL/WITHOUT/EMBEDDINGS/COLUMN")
     pipeline = Pipeline().setStages([
        embeddings,
        graph_builder,
        ner_approach
    ])
    trained_model = pipeline.fit(training_dataset)
```

## Context
Confusing developer experience. Forces having to transform test dataset to have embeddings and be saved as parquet, rather than allowing the CoNLL file to be in whatever format (plain text, avro, delta, etc).

### Spark NLP version `sparknlp.version()`:
4.2.1
### Apache Spark version `spark.version`:
3.3.1
### Java version `java -version`:
openjdk version "1.8.0_292"
OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_292-b10)
OpenJDK 64-Bit Server VM (AdoptOpenJDK)(build 25.292-b10, mixed mode)
### Setup and installation (Pypi, Conda, Maven, etc.):
PyPi
### Operating System and version:
  System Version:	macOS 12.6 (21G115)
  Kernel Version:	Darwin 21.6.0
  Boot Volume:	Macintosh HD
  Boot Mode:	Normal

**Comment by maziyarpanahi** [comment]
Hi @ethnhll 

>I would expect to be able to provide a path to a plain-text CoNLL file for a test dataset without failure.
That's not how this parameter was designed/implemented. The path must point to a saved Parquet that has already been transformed with the exact same Word Embeddings (pipeline) the NerDLApproach needs to be trained. This parameter has the exact same behavior as other trainable annotators where they have `testDataset` parameter. (this is the intended use, this parameter is not required for the training, and it is the only way we can use that for internal_prediction/metrics - we cannot accept a raw CoNLL file here, you are not even using a raw CoNLL file for training the NER)

I will add this as a documentation issue (even though we have many examples for this in `spark-nlp-workshop` so it's not a totally unknown feature). But we will leave an example of how to prepare the test dataset.

@DevinTDHa Could you please have an example of how to prepare `testDataset` for NerDLApproach, ClassifierDLApproach, SentimentDLApproach, and MultiClassifierDLApproach here? https://nlp.johnsnowlabs.com/docs/en/training#nerdlapproach

**Comment by ethnhll** [comment]
I can understand that it wasn't designed that way, and this being a documentation issue is fine. required embeddings on a test set for NER seems unnecessary, but if it is documented as a requirement for the test dataset, then that is fine with me.

**Comment by ethnhll** [comment]
For further clarification, is the test dataset required to be in Parquet only, or can it be in Avro or Delta also?

**Comment by maziyarpanahi** [comment]
We didn't have much of a choice when we added that feature (still the same)

- A raw CoNLL file needs to be transformed to have the word embeddings for prediction to which we don't have access to word embeddings inside the NerDLApproach. Stages inside Pipeline don't have access to previous stages in the Pipeline. (this comes from Spark ML)
- We also cannot pass a DataFrame inside Spark param (testDataset), this also comes from Spark ML serialization

The only option we had to provide that feature (which was requested) was to have the CoNLL loaded, transformed, and written as Parquet so we can read it normally inside NerDLApproach by simply passing the Path. The format `parquet` seemed to be supported on almost all file systems (locally and in distributed file systems). If today that is not true, we may be able to add another format under the conditions that first, it is simply possible to be loaded back as `spark.read.format("")`. Second, it doesn't require any external/third-party dependencies and it is natively supported by Apache Spark. 

I will have a look at `delta` to see if we can save/load our DataFrame without any issues there.
